Ziiig

sakaadar opened the pull request:
Both Servers done, have a look, extracted some methods to a Util class.

sakaadar committed:
DirectoryTcpServer, altmost done needs to use the helper functions created, should be finished soon

sakaadar committed:
DirectoryTcpServer implemented to handle request from Vinyl servers "REGISTER/UPDATE"

sakaadar committed:
DirectoryTcpServer finished, DirectoryUDP still in progress

sakaadar committed:
Implemented Rest of UDPServer, implemented the handler method for UDP request, set some strict requirements and exepcition handling. Testing will be done soon.
- Extracted some functions to a Util class, because both servers needed to use the same logic some times

Ghiyathali reviewed (commented):
I will take a look, im also almost done with mine , im just trying to polish the code and make sure it follows the constraints in the word doc.

sakaadar committed:
Moved some helper functions to the util class, because we had to use it in the Server layer.

sakaadar committed:
Moved some stuff around, and added a model layer to seperate dependencies, created a Track class to simulate actual Vinyls.

sakaadar committed:
In CatalogServer the handleClient method was - fully implemented, added a helper function, - to find the track by id. alså changed the - structure of the function writeJsonLine, to take a arbitraty parameter in the method.

sakaadar committed:
Forgot stop method

sakaadar committed:
Many modifications done today.
- Systemintegration for the whole system tried a manuel testing, from the Directory and so on, but as of now it, when the server connects to the Directory it breaks, the message is received but there isnt any feedback from the directory. Client has not been tested yet but implemented.

sakaadar committed:
Directorymain works and accepts connection from server, Servermain connects to directory through TCP and the client can lookup a server in the directory.

- As of now LIST, GET, SEARCH Doesnt work as intented. gives a error message every time you try to access.

sakaadar committed:
Up to date, the system works for the directory, client and Server works as intented

- Client -> Directory (client asks for a server name in return it gets the name,ip and TTL)
- Client -> Server (Client can request for list of tracks with artist,song,date)
-Client -> Server (Client can search for the name of a artist and in return get the track)
-Client -> Server (Client get request a track using get)
-Server -> Directory (Establish TCP connection and Server registers with the default server name "Happy_music.group3.pro2")
-Server -> Directory (Renew its TCP Connection every 7 minutes)
